packages/ketcher-react/src/script/ui/views/components/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,182 @@
+/****************************************************************************
+ * Copyright 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+import { Action, fromBondsAttrs, fromOneBondDeletion } from 'ketcher-core'
+import { useCallback } from 'react'
+import type { ItemParams, PredicateParams } from 'react-contexify'
+import { Item, Menu, Submenu } from 'react-contexify'
+import 'react-contexify/ReactContexify.css'
+import { useAppContext } from 'src/hooks'
+import Editor from 'src/script/editor'
+import tools from 'src/script/ui/action/tools'
+import Icon from 'src/script/ui/component/view/icon'
+import styles from './ContextMenu.module.less'
+
+export const CONTEXT_MENU_ID = 'KeTcHeR-CoNtExT-MeNu'
+const bondTools = Object.keys(tools).filter((key) => key.startsWith('bond-'))
+
+export interface ContextMenuItemProps {
+  selected: boolean
+  ci: any
+}
+type ItemData = unknown
+
+const ContextMenu: React.FC = () => {
+  const { getKetcherInstance } = useAppContext()
+
+  const handleEdit = useCallback(
+    async ({ props }: ItemParams<ContextMenuItemProps, ItemData>) => {
+      const editor = getKetcherInstance().editor as Editor
+      const bondId = props?.ci.id
+      const bond = editor.render.ctab.bonds.get(bondId)?.b
+
+      try {
+        const newBond = await editor.event.bondEdit.dispatch(bond)
```

Comment:
What kind of error may occur?

Reply:
![image](https://user-images.githubusercontent.com/27288153/206634747-7e2a80b9-2472-469d-a23d-8504434153b0.png)

If clicking the Cancel button, will throw an error `undefined`. What I did is catch it and `console.error` it.

I searched `bondEdit.dispatch(bond)` in the codebase to see how they handled it. I got this:
```js
catch(() => null) // w/o changes
``` 
So maybe it's better to ignore it as they did? What's your take on that

Reply:
Yes, you can do the same way. Or you can make it more explicit, for example:
```
const noOperation = () => {}
catch(noOperation)
```